cassiopeia/core/tft_summoner.py

```diff
@@ -0,0 +1,173 @@
+import arrow
+import datetime
+from typing import Union
+
+from datapipelines import NotFoundError
+from merakicommons.cache import lazy_property
+from merakicommons.container import searchable
+
+from ..data import Region, Platform, Rank
+from .common import CoreData, CassiopeiaObject, CassiopeiaGhost, provide_default_region, ghost_load_on
+from .staticdata import ProfileIcon
+from ..dto.tft_summoner import TFTSummonerDto
+
+
+##############
+# Data Types #
+##############
+
+
+class TFTSummonerData(CoreData):
```

Comment:
TFTSummoner and LeagueSummoner is the same object. The call url is the only difference

Reply:
Yeah, they're definitely pretty similar (with the exception of the section below `# Special core methods`, which will be different between the two). My thinking here was that while _/lol/summoner/v4_ and _/tft/summoner/v1_ are currently identical, they might evolve independently of each other and return subtly different DTOs in future. The duplication felt incidental for now

Would you prefer to just use Summoner, and put TFT helper methods into there e.g. `Summoner(name='Boatshoes').tft_match_history` and handle unpicking the two if/when changes happen?

Reply:
What about inheriting from the original summoner object instead of CoreData?

Reply:
This is a good point. We might side-step it all together by the comment I am about to make below.